optking/v3d.py

```diff
@@ -178,34 +178,36 @@ def tors(A, B, C, D):
         EBA = eAB(B, A)
         EAB = -1 * EBA
     except AlgError as error:
-        logger.warning("Could not normalize %d, %d vector in tors()\n" % (str(A), str(B)))
+        logger.warning("Could not normalize {}, {} vector in tors()\n".format(str(A), str(B)))
```

Comment:
similarly, `f"Could not normalize {A}, {B} vector in tors()\n"`

Reply:
That's nice.  When will the python folks finish coming up with new ways to print :)  So f" looks good to use when you are confident going with a natural formatting.

Reply:
Yes, everyone seems to like f-strings, even those who couldn't decide between the `%` way and the `.format` way. You can even put expressions inside `f'{title.upper()}'`, `f"my full name is {names['LAB']}"`. Big caveat is they're py 3.6+, but that's true for everything in psi4 and qca ecosystems, so no problem.